src/yang/ietf-access-control-list.yang

```diff
@@ -422,145 +457,132 @@ module ietf-access-control-list {
-                   "'acl:mixed-l2-l3-ipv4-acl')";
-              uses packet-fields:acl-eth-header-fields;
-              uses packet-fields:acl-ip-header-fields;
-              uses packet-fields:acl-ipv4-header-fields;
-              description
-                "Rule set that is a logical AND (&&) of l2 
-                 and ipv4 headers.";
-            }
-
-            container l2-l3-ipv6-acl {
-              if-feature mixed-ipv6-acl;
-              must "derived-from(../../../../acl-type, " +
-                   "'acl:mixed-l2-l3-ipv6-acl')";
-              uses packet-fields:acl-eth-header-fields;
-              uses packet-fields:acl-ip-header-fields;
-              uses packet-fields:acl-ipv6-header-fields;
-              description
-                "Rule set that is a logical AND (&&) of L2 
-                 && IPv6 headers.";
-            }
-
-            container l2-l3-ipv4-ipv6-acl {
-              if-feature l2-l3-ipv4-ipv6-acl;
-              must "derived-from(../../../../acl-type, " +
-                   "'acl:mixed-l2-l3-ipv4-ipv6-acl')";
-              uses packet-fields:acl-eth-header-fields;
-              uses packet-fields:acl-ip-header-fields;
-              uses packet-fields:acl-ipv4-header-fields;
-              uses packet-fields:acl-ipv6-header-fields;
-              description
-                "Rule set that is a logical AND (&&) of L2 
-                 && IPv4 && IPv6 headers.";
-            }
-            
-            container tcp-acl {
-              if-feature tcp-acl;
+            container tcp {
+              if-feature match-on-tcp;
               uses packet-fields:acl-tcp-header-fields;
               description
                 "Rule set that defines TCP headers.";
             }
-            
-            container udp-acl {
-              if-feature udp-acl;
+
+            container udp {
+              if-feature match-on-udp;
               uses packet-fields:acl-udp-header-fields;
               description
                 "Rule set that defines UDP headers.";
             }
-            
-            container icmp-acl {
-              if-feature icmp-acl;
+
+            container icmp {
+              if-feature match-on-icmp;
               uses packet-fields:acl-icmp-header-fields;
               description
                 "Rule set that defines ICMP headers.";
             }
-             
-            container any-acl {
-              if-feature any-acl;
-              must "derived-from(../../../../acl-type, 'acl:any-acl')";
-              presence "Matches any";
+
+            leaf any {
+              when "derived-from(../../../../acl-type, " +
+                   "'acl:any-acl-type')";
+              type empty;
               description
                 "Rule set that allows for a any ACL.";
             }
-            
-            leaf interface {
-              type if:interface-ref;
+
+            choice attachment-point {
```

Comment:
I think this is wrong. The attachment-point is different. This is about matching on ingress/egress interface in an ACE.

Trying to match on egress-interface if the ACL is then attached to a interface different from the one being matched upon is nonsensical but we don't try to prevent it through YANG. Just add egress-interface and ingress-interface leaves, just like I did in my foo branch.